cloud/src/autogluon/cloud/utils/iam.py

```diff
@@ -0,0 +1,295 @@
+                "s3:ListBucket",
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketCors",
+                "s3:PutBucketCors"
+            ],
+            "Resource": [
+                "arn:aws:s3:::*"
+            ]
+        },
+        {
+            "Sid": "ListEvents",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "sagemaker:ListEndpointConfigs",
+                "sagemaker:ListEndpoints",
+                "sagemaker:ListTransformJobs",
+                "sagemaker:ListTrainingJobs",
+                "sagemaker:ListModels"
+            ],
+            "Resource": [
+                "*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "sagemaker:CreatePresignedDomainUrl",
+                "sagemaker:DescribeDomain",
+                "sagemaker:ListDomains",
+                "sagemaker:DescribeUserProfile",
+                "sagemaker:ListUserProfiles",
+                "sagemaker:*App",
+                "sagemaker:ListApps"
+            ],
+            "Resource": [
+                "*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": "sagemaker:*",
+            "Resource": [
+                "arn:aws:sagemaker:*:*:flow-definition/*"
+            ],
+            "Condition": {
+                "StringEqualsIfExists": {
+                    "sagemaker:WorkteamType": [
+                        "private-crowd",
+                        "vendor-crowd"
+                    ]
+                }
+            }
+        },
+        {
+            "Sid": "Others",
+            "Effect": "Allow",
+            "Action": [
+                "application-autoscaling:DeleteScalingPolicy",
+                "application-autoscaling:DeleteScheduledAction",
+                "application-autoscaling:DeregisterScalableTarget",
+                "application-autoscaling:DescribeScalableTargets",
+                "application-autoscaling:DescribeScalingActivities",
+                "application-autoscaling:DescribeScalingPolicies",
+                "application-autoscaling:DescribeScheduledActions",
+                "application-autoscaling:PutScalingPolicy",
+                "application-autoscaling:PutScheduledAction",
+                "application-autoscaling:RegisterScalableTarget",
+                "cloudwatch:DeleteAlarms",
+                "cloudwatch:DescribeAlarms",
+                "cloudwatch:GetMetricData",
+                "cloudwatch:GetMetricStatistics",
+                "cloudwatch:ListMetrics",
+                "cloudwatch:PutMetricAlarm",
+                "cloudwatch:PutMetricData",
+                "ec2:CreateNetworkInterface",
+                "ec2:CreateNetworkInterfacePermission",
+                "ec2:CreateVpcEndpoint",
+                "ec2:DeleteNetworkInterface",
+                "ec2:DeleteNetworkInterfacePermission",
```

Comment:
Do we need network modification permissions?

Reply:
Technically, the training/deploy etc api call to api support vpc settings

Reply:
What is the VPC being used for? Do we need VPC for training and inference endpoint? We can just enable network isolation to block the internet access if needed: https://docs.amazonaws.cn/en_us/sagemaker/latest/dg/mkt-algo-model-internet-free.html

Reply:
so sagemaker api allow user to pass vpc stuffs. After discussion with Alex, we agree that we shouldn't put this in the policy we provided to the user. User can add it themselves if they need so

Reply:
@tonyhoo VPC/subnets/security groups should be created outside of any SM activity, they should just be passed to the call as the existing resources.

Reply:
I know some users might have VPC set up which enhance the data security. However, wondering if AG users need it as the container won't actively polling the data outside of the container env. Am I right? If so, we can set the network_isolation flag to True to prevent any internet access for security